From 5a109109e4fe052094ac38590e10f946ab5772ee Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 14 Sep 2017 14:43:10 +0800
Subject: [PATCH 437/895] drm: fence: replace dma_fence with fence

Since drop dma_fence because there are two much changes for the whole system.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/drm_atomic.c        |    6 +++---
 drivers/gpu/drm/drm_drv.c           |    1 +
 drivers/gpu/drm/drm_fb_cma_helper.c |    2 +-
 include/drm/drm_atomic.h            |    2 +-
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic.c b/drivers/gpu/drm/drm_atomic.c
index 30f3cc4..a305965 100644
--- a/drivers/gpu/drm/drm_atomic.c
+++ b/drivers/gpu/drm/drm_atomic.c
@@ -1217,7 +1217,7 @@ EXPORT_SYMBOL(drm_atomic_set_fb_for_plane);
 /**
  * drm_atomic_set_fence_for_plane - set fence for plane
  * @plane_state: atomic state object for the plane
- * @fence: dma_fence to use for the plane
+ * @fence: fence to use for the plane
  *
  * Helper to setup the plane_state fence in case it is not set yet.
  * By using this drivers doesn't need to worry if the user choose
@@ -1233,10 +1233,10 @@ EXPORT_SYMBOL(drm_atomic_set_fb_for_plane);
  */
 void
 drm_atomic_set_fence_for_plane(struct drm_plane_state *plane_state,
-			       struct dma_fence *fence)
+			       struct fence *fence)
 {
 	if (plane_state->fence) {
-		dma_fence_put(fence);
+		fence_put(fence);
 		return;
 	}
 
diff --git a/drivers/gpu/drm/drm_drv.c b/drivers/gpu/drm/drm_drv.c
index 9abb49a..f5ea09b 100644
--- a/drivers/gpu/drm/drm_drv.c
+++ b/drivers/gpu/drm/drm_drv.c
@@ -295,6 +295,7 @@ void drm_minor_release(struct drm_minor *minor)
 	drm_dev_unref(minor->dev);
 }
 
+int drm_dev_set_unique(struct drm_device *dev, const char *name);
 /**
  * DOC: driver instance overview
  *
diff --git a/drivers/gpu/drm/drm_fb_cma_helper.c b/drivers/gpu/drm/drm_fb_cma_helper.c
index 81d6899..e54c31b 100644
--- a/drivers/gpu/drm/drm_fb_cma_helper.c
+++ b/drivers/gpu/drm/drm_fb_cma_helper.c
@@ -285,7 +285,7 @@ int drm_fb_cma_prepare_fb(struct drm_plane *plane,
 			  struct drm_plane_state *state)
 {
 	struct dma_buf *dma_buf;
-	struct dma_fence *fence;
+	struct fence *fence;
 
 	if ((plane->state->fb == state->fb) || !state->fb)
 		return 0;
diff --git a/include/drm/drm_atomic.h b/include/drm/drm_atomic.h
index 5dfbe48..3210bda 100644
--- a/include/drm/drm_atomic.h
+++ b/include/drm/drm_atomic.h
@@ -346,7 +346,7 @@ drm_atomic_set_crtc_for_plane(struct drm_plane_state *plane_state,
 void drm_atomic_set_fb_for_plane(struct drm_plane_state *plane_state,
 				 struct drm_framebuffer *fb);
 void drm_atomic_set_fence_for_plane(struct drm_plane_state *plane_state,
-				    struct dma_fence *fence);
+				    struct fence *fence);
 int __must_check
 drm_atomic_set_crtc_for_connector(struct drm_connector_state *conn_state,
 				  struct drm_crtc *crtc);
-- 
1.7.5.4

